components/console-backend-service/internal/domain/k8s/namespace_resolver_test.go

```diff
@@ -0,0 +1,147 @@
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+)
+
+func TestNamespaceResolver_NamespacesQuery(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		name := "name"
+		labels := map[string]string{
+			"env": "true",
+		}
+		resource := fixNamespace(name, labels)
+		resources := []*v1.Namespace{resource, resource}
+		expectedResult := gqlschema.Namespace{
+			Name:   name,
+			Labels: labels,
+		}
+
+		expected := []gqlschema.Namespace{
+			expectedResult, expectedResult,
+		}
+		svc := automock.NewNamespaceSvc()
+		appRetriever := new(appAutomock.ApplicationRetriever)
+		svc.On("List").Return(resources, nil).Once()
+		defer svc.AssertExpectations(t)
+
+		converter := automock.NewNamespaceConverter()
+		converter.On("ToGQLs", resources).Return(expected, nil).Once()
+		defer converter.AssertExpectations(t)
+
+		resolver := k8s.NewNamespaceResolver(svc, appRetriever)
+		resolver.SetNamespaceConverter(converter)
+
+		result, err := resolver.NamespacesQuery(nil, nil)
+
+		require.NoError(t, err)
+		assert.Equal(t, expected, result)
+	})
+
+	t.Run("Empty", func(t *testing.T) {
+		resources := []*v1.Namespace{}
+		expected := []gqlschema.Namespace(nil)
+		svc := automock.NewNamespaceSvc()
+		appRetriever := new(appAutomock.ApplicationRetriever)
+		svc.On("List").Return(resources, nil).Once()
+		defer svc.AssertExpectations(t)
+
+		converter := automock.NewNamespaceConverter()
+		converter.On("ToGQLs", resources).Return(expected, nil).Once()
+		defer converter.AssertExpectations(t)
+
+		resolver := k8s.NewNamespaceResolver(svc, appRetriever)
+		resolver.SetNamespaceConverter(converter)
+
+		result, err := resolver.NamespacesQuery(nil, nil)
+
+		require.NoError(t, err)
+		assert.Equal(t, expected, result)
+	})
+
+	t.Run("Error", func(t *testing.T) {
+		svc := automock.NewNamespaceSvc()
+		appRetriever := new(appAutomock.ApplicationRetriever)
+		svc.On("List").Return(nil, errors.New("Error")).Once()
+		defer svc.AssertExpectations(t)
+		resolver := k8s.NewNamespaceResolver(svc, appRetriever)
+
+		result, err := resolver.NamespacesQuery(nil, nil)
+
+		require.Error(t, err)
+		assert.True(t, gqlerror.IsInternal(err))
+		assert.Nil(t, result)
+	})
+
+	t.Run("ErrorConverting", func(t *testing.T) {
+		resources := []*v1.Namespace{}
+		expected := errors.New("Test")
+		svc := automock.NewNamespaceSvc()
+		appRetriever := new(appAutomock.ApplicationRetriever)
+		svc.On("List").Return(resources, nil).Once()
+		defer svc.AssertExpectations(t)
+
+		converter := automock.NewNamespaceConverter()
+		converter.On("ToGQLs", resources).Return(nil, expected).Once()
+		defer converter.AssertExpectations(t)
+
+		resolver := k8s.NewNamespaceResolver(svc, appRetriever)
+		resolver.SetNamespaceConverter(converter)
+
+		result, err := resolver.NamespacesQuery(nil, nil)
+
+		require.Error(t, err)
+		assert.True(t, gqlerror.IsInternal(err))
+		assert.Nil(t, result)
+	})
+}
+
+func TestNamespaceResolver_CreateNamespaceMutationn(t *testing.T) {
```

Comment:
As it is the requirement for now I'd add a test to verify if `env=true` is added even if it is set to false or missing.